src/librustc_resolve/lib.rs

```diff
@@ -3457,6 +3508,107 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         found_traits
     }
 
+    /// When name resolution fails, this method can be used to look up candidate
+    /// entities with the expected name. It allows filtering them using the
+    /// supplied predicate (which should be used to only accept the types of
+    /// definitions expected e.g. traits). The lookup spans across all crates.
+    ///
+    /// NOTE: The method does not look into imports, so it ignore all private-
+    /// made-public entities, such as the following code pattern:
+    /// ```
+    /// pub mod foo {
+    ///     mod bar { trait T {}; };
+    ///     // make private T accessible through an import:
+    ///     pub use bar::T as T;
+    /// }
```

Comment:
That simplifies things a lot. I was under the impression that you can "reveal" private parts of a crate through `pub use` statements. If imports are only visible on the current crate, we're perfectly fine ignoring them :+1: 

Thanks a lot for explaining this!